base/stacktraces.jl

```diff
@@ -137,16 +137,26 @@ function lookup(ip::Base.InterpreterIP)
         line = meth.line
         codeinfo = meth.source
     else
+        func = top_level_scope_sym
+        file = empty_sym
+        line = Int32(0)
         if code isa Core.CodeInstance
             codeinfo = code.inferred::CodeInfo
+            def = code.def
+            if isa(def, Core.ABIOverride)
+                def = def.def
+            end
```

Comment:
The arbitrary proliferation of these checks seems a bad sign, but that is unrelated to this PR (which looks good to me)

Reply:
Yeah, we just need to put these behind a better accessor. I'll clean that up in the future.